### PR TITLE
azuki: init at 0-unstable-2021-07-02

### PR DESCRIPTION
**Pull request by nyadiia**
## Description of changes

azuki is a monospace handwritten font with a few variants:
- P - proportional ( a normal font )
- L - no emojis
- B - bold monospace
- LP - proportional with no emojis
- LB - bold monospace with no emojis

[azukifont.com/font/azuki.html](http://azukifont.com/font/azuki.html)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/az/azuki-b/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  fetchzip,
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "azuki-b";
+  version = "1.20";
+  downloadVersion = "B120";
+
+  src = fetchzip {
+    url = "https://azukifont.com/font/azukifont${downloadVersion}.zip";
```
Comment (Aleksanaa) [inline pkgs/by-name/az/azuki-b/package.nix:13]: They are all from the same source and installing the same files? If it’s not big, there’s actually no need to unpack it.

File: pkgs/by-name/az/azuki-b/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    runHook preInstall
+
+    install -Dm644 azukifont${downloadVersion}/*.ttf -t $out/share/fonts/truetype
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "http://azukifont.com/font/azuki.html";
+    description = "Azuki Monospace Font Bold with Emoji";
```
Comment (Aleksanaa) [inline pkgs/by-name/az/azuki-b/package.nix:28]: missing `license`

File: pkgs/by-name/az/azuki-b/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm644 azukifont${downloadVersion}/*.ttf -t $out/share/fonts/truetype
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/az/azuki-b/package.nix:26]: Better to avoid `with lib`

**Comment by nyadiia** [comment]
rewrote this package, fixing the different version across all variations by just using the date of the latest one. all variations have been put into the same azuki package. license is unfree because the [terms of use on the site](https://azukifont.com/kiyaku.html) says no redistribution.

File: pkgs/by-name/az/azuki/package.nix [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = {
+    lib.homepage = "http://azukifont.com/font/azuki.html";
+    lib.description = "Azuki Font";
+    lib.license = lib.licenses.unfree;
+    lib.platforms = lib.platforms.all;
+    lib.maintainers = with lib.maintainers; [ nyadiia ];
```
[the remark this answers is not in the document]
Comment (nyadiia) [reply pkgs/by-name/az/azuki/package.nix:76]: i'm not using `with lib;` at the beginning ( may add it back tho i don't see a huge issue with it )

**Comment by nyadiia** [comment]
actually i am going to use `with lib;` i don't see an issue with it and it makes it look far nicer

**Comment by nyadiia** [comment]
ok now things should be good :)

**Review by remexre — APPROVED** [review]

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
A bit of pedantry, but otherwise this looks good.

**Review by tomodachi94 — APPROVED** [review]
Lgtm. :heart:

**Review by remexre — APPROVED** [review]

**Comment by tomodachi94** [comment]
Merged, thank you for your patience 💖

**Comment by nyadiia** [comment]
thank you so much! <3

**Pull request merged**